File: NSKButtonObjC/NSKButtonObjC/DecodeTestViewController.m
//
//  DecodeTestViewController.m
//  NSKButtonObjC
//
//  Created by NSSimpleApps on 24.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

#import "DecodeTestViewController.h"
#import "NSKButton.h"

@interface DecodeTestViewController ()

@end

@implementation DecodeTestViewController

- (void)viewDidLoad {
    
    [super viewDidLoad];
    
    self.title = @"Decode NSKButton";
    
    id object = [NSKeyedUnarchiver unarchiveObjectWithData:self.data];
    
    if (object && [object isKindOfClass:[NSKButton class]]) {
        
        NSKButton *button = object;
        button.center = self.view.center;
        
        [self.view addSubview:button];
        
        NSLayoutConstraint *horizontalConstraint = [button.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor];
        NSLayoutConstraint *vertivalConstraint = [button.centerYAnchor constraintEqualToAnchor:self.view.centerYAnchor];
        
        [NSLayoutConstraint activateConstraints:@[horizontalConstraint, vertivalConstraint]];
    }
    
}


@end

File: NSKButtonSwift/NSKButtonSwift/DecodeTestViewController.swift
//
//  DecodeTestViewController.swift
//  NSKButtonSwift
//
//  Created by NSSimpleApps on 25.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

import UIKit

class DecodeTestViewController: UIViewController {
    
    var data: Data!
    
    override func viewDidLoad() {
        
        super.viewDidLoad()

        self.title = "Decode NSKButton"
        
        if let button = NSKeyedUnarchiver.unarchiveObject(with: self.data) as? NSKButton {
            
            button.center = self.view.center
            
            self.view.addSubview(button)
            
            NSLayoutConstraint(item: button, attribute: .centerX, relatedBy: .equal, toItem: self.view, attribute: .centerX, multiplier: 1, constant: 0).isActive = true
            NSLayoutConstraint(item: button, attribute: .centerY, relatedBy: .equal, toItem: self.view, attribute: .centerY, multiplier: 1, constant: 0).isActive = true
        }
    }
}
